crates/sc-consensus-subspace-rpc/src/lib.rs

```diff
@@ -798,4 +810,25 @@ where
 
         Ok(last_segment_headers)
     }
+
+    // TODO:
+    // - the number of object mappings in each segment can be very large (hundreds or thousands)
+    //   to avoid RPC connection failures, limit the number of mappings returned in each response,
+    //   or the number of in-flight responses
+    // - provide mappings from the genesis segment, using recreate_genesis_segment()
```

Comment:
I don't think we have or will have mappings in genesis segment to worry about this.

Reply:
Turned into a comment on the code in 95f6b8ca03, along with the changes to turn the iterator into a GlobalObject mapping.